src/NServiceBus.Core/Receiving/ProcessingCompleted.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Event raised when the transport has completed processing a message.
+    /// </summary>
+    public class ProcessingCompleted
+    {
+        /// <summary>
+        /// Constructs the event.
+        /// </summary>
+        public ProcessingCompleted(string messageId, bool wasAcknowledged, Dictionary<string, string> headers, DateTimeOffset startedAt, DateTimeOffset completedAt)
+        {
+            MessageId = messageId;
+            WasAcknowledged = wasAcknowledged;
+            Headers = headers;
+            StartedAt = startedAt;
+            CompletedAt = completedAt;
+        }
+
+        /// <summary>
+        /// The ID of the message.
+        /// </summary>
+        public string MessageId;
+
+        /// <summary>
+        /// True if the message was acknowledged and removed from the queue.
+        /// </summary>
+        public bool WasAcknowledged;
+
+        /// <summary>
+        /// The message headers.
+        /// </summary>
+        public Dictionary<string, string> Headers { get; }
```

Comment:
Why is this required?

Reply:
Since metrics and other users of this will need to access things in the headers. Metrics needs the sent time header and the message type header